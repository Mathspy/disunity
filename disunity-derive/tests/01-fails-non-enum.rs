use disunity_derive::Variant;

#[derive(Variant)]
struct Class {
    #[disunity(discriminate = 1)]
    field: i32,
}

#[derive(Variant)]
union MyUnion {
    #[disunity(discriminate = 1)]
    f1: u32,
    #[disunity(discriminate = 2)]
    f2: f32,
}

fn main() {}
