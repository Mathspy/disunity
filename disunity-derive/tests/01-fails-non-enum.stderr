error: Variant can't be derived on structs
 --> tests/01-fails-non-enum.rs:4:1
  |
4 | struct Class {
  | ^^^^^^

error: Variant can't be derived on unions
  --> tests/01-fails-non-enum.rs:10:1
   |
10 | union MyUnion {
   | ^^^^^
